src/screens/DiscoveryScreenV2/elements/Search.js

```diff
@@ -134,6 +133,14 @@ const DiscoverySearch = ({
     debounceChangeText(searchText);
   }, [searchText]);
 
+  React.useEffect(() => {
+    if (discoverySearchBarRef?.current) {
+      setTimeout(() => {
+        discoverySearchBarRef?.current?.focus();
+      }, 500);
+    }
+  }, []);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This new `useEffect` hook is used to focus on the search bar after a delay of 500ms. This is a good approach to ensure that the search bar gets focused even if it wasn't ready at the time of initial render. However, please make sure that the delay doesn't negatively impact the user experience.

```javascript
React.useEffect(() => {
  if (discoverySearchBarRef?.current) {
    setTimeout(() => {
      discoverySearchBarRef?.current?.focus();
    }, 500);
  }
}, []);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->